src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/Requests/DataSetQueryCriteria.cs

```diff
@@ -1,3 +1,19 @@
+using System.Text.Json.Serialization;
+using FluentValidation.Validators;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Requests.Converters;
+
 namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Requests;
 
-public abstract record DataSetQueryCriteria;
+[JsonConverter(typeof(DataSetQueryCriteriaJsonConverter))]
+public abstract record DataSetQueryCriteria
+{
+    protected static void InheritanceValidator<TCriteria>(
+        PolymorphicValidator<TCriteria, DataSetQueryCriteria> validator)
+        where TCriteria : DataSetQueryCriteria
+    {
+        validator.Add(_ => new DataSetQueryCriteriaAnd.Validator());
```

Comment:
Can this just be `validator.Add(new DataSetQueryCriteriaAnd.Validator());`?

Same with the other lines

Reply:
Nope, we need to defer this or we run into a recursive loop which crashes the server.